src/main/java/uk/gov/hmcts/reform/migration/CaseMigrationProcessor.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.hmcts.reform.migration;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.domain.exception.CaseMigrationException;
+import uk.gov.hmcts.reform.migration.ccd.CoreCaseDataService;
+import uk.gov.hmcts.reform.migration.repository.ElasticSearchRepository;
+import uk.gov.hmcts.reform.migration.repository.IdamRepository;
+import uk.gov.hmcts.reform.migration.service.DataMigrationService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Component
+public class CaseMigrationProcessor {
+    private static final String EVENT_ID = "migrateCase";
+    private static final String EVENT_SUMMARY = "Migrate Case";
+    private static final String EVENT_DESCRIPTION = "Migrate Case";
+    public static final String LOG_STRING = "-----------------------------------------";
+
+    @Autowired
+    private CoreCaseDataService coreCaseDataService;
+
+    @Autowired
+    private DataMigrationService<Map<String, Object>> dataMigrationService;
+
+    @Autowired
+    private ElasticSearchRepository elasticSearchRepository;
+
+    @Autowired
+    private IdamRepository idamRepository;
+
+    @Getter
+    private List<Long> migratedCases = new ArrayList<>();
+
+    @Getter
+    private List<Long> failedCases = new ArrayList<>();
+
+    @Value("${case-migration.processing.limit}")
+    private int caseProcessLimit;
+
+    public void migrateCases(String caseType) {
+        validateCaseType(caseType);
+        log.info("Data migration of all cases started for case type: {}", caseType);
```

Comment:
@kiran-yenigala-hmcts We can't exactly say all cases will be considered for migration as we have a processing limit now. So, the logger statement can be modified to highlight that.

Reply:
removed **all** from this statement